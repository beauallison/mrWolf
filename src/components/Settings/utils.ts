import * as utils from '../../utils';
import ICity from '../../ICity';

// Used specifically for React Select
export interface IValue {
  value: string;
  label: string;
  city: ICity;
}

// City to Value
export const cityToValue = (city: ICity) => {
  const nameAndCountry = `${city.name}, ${city.country}`;

  return {
    value: nameAndCountry,
    label: nameAndCountry,
    city,
  } as IValue;
};

export const citiesToValues = (cities: ICity[]) => cities.map((city) => cityToValue(city));

export const generateOptions = () => citiesToValues(utils.citiesList());

// Value to City
export const valueToCity = ({ city }: IValue) => city as ICity;
export const valuesToCities = (values: IValue[]) => values.map((value) => valueToCity(value));
