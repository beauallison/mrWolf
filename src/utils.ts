import _ from 'lodash';
import Cities from './data/cities.json';
import Countries from './data/countries.json';
import ICity from './ICity';
import moment from 'moment-timezone';

export const citiesList = (): ICity[] =>
  Cities.map((city: any) => {
    const { code } = city;
    const country = _.find(Countries, { code });
    return {
      name: city.name,
      timezone: city.tz,
      country: country && country.name,
    } as ICity;
  });

export interface IGenerateTime {
  timezone: string;
  display24HourTime?: boolean;
  displaySeconds?: boolean;
  displayPeriod?: boolean;
}

export const generateTime = (props: IGenerateTime) => {
  const hourFormat = props.display24HourTime ? 'HH:mm' : 'h:mm';
  const periodFormat = props.displayPeriod && !props.display24HourTime ? `${hourFormat}a` : hourFormat;
  const format = props.displaySeconds ? `${hourFormat}:ss ` : periodFormat;
  return moment().tz(props.timezone).format(format);
};
