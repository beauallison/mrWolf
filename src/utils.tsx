import _ from 'lodash';
import Cities from './data/cities.json';
import Countries from './data/countries.json';
import ICity from './ICity';
import moment from 'moment-timezone';

export const citiesList = (): ICity[] =>
  Cities.map((city: any) => {
    const { code } = city;
    const country = _.find(Countries, { code });
    return {
      name: city.name,
      timezone: city.tz,
      country: country && country.name,
    } as ICity;
  });

export interface IGenerateTime {
  timezone: string;
  displaySeconds?: boolean;
}

export const generateTime = (props: IGenerateTime) => {
  const format = props.displaySeconds ? 'HH:MM:ss' : 'HH:MM';
  return moment().tz(props.timezone).format(format);
};
