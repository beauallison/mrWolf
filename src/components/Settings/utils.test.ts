import { cityToValue, citiesToValues, generateOptions, valueToCity, valuesToCities } from './utils';

describe('utils', () => {
  const city = {
    country: 'United Arab Emirates',
    name: 'Dubai',
    timezone: 'Asia/Dubai',
  };

  const value = {
    city,
    label: 'Dubai, United Arab Emirates',
    value: 'Dubai',
  };

  it('should generate city options', () => {
    const options = generateOptions();
    expect(options.length).toEqual(6184);
    expect(options[0]).toMatchSnapshot();
  });

  it('should convert cities to value', () => {
    expect(cityToValue(city)).toMatchSnapshot();
    expect(citiesToValues([city, city])).toMatchSnapshot();
  });

  it('should convert values to cities', () => {
    expect(valueToCity(value)).toMatchSnapshot();
    expect(valuesToCities([value, value])).toMatchSnapshot();
  });
});
