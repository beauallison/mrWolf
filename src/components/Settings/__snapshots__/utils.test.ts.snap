// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils should convert cities to value 1`] = `
Object {
  "city": Object {
    "country": "United Arab Emirates",
    "name": "Dubai",
    "timezone": "Asia/Dubai",
  },
  "label": "Dubai, United Arab Emirates",
  "value": "Dubai, United Arab Emirates",
}
`;

exports[`utils should convert cities to value 2`] = `
Array [
  Object {
    "city": Object {
      "country": "United Arab Emirates",
      "name": "Dubai",
      "timezone": "Asia/Dubai",
    },
    "label": "Dubai, United Arab Emirates",
    "value": "Dubai, United Arab Emirates",
  },
  Object {
    "city": Object {
      "country": "United Arab Emirates",
      "name": "Dubai",
      "timezone": "Asia/Dubai",
    },
    "label": "Dubai, United Arab Emirates",
    "value": "Dubai, United Arab Emirates",
  },
]
`;

exports[`utils should convert values to cities 1`] = `
Object {
  "country": "United Arab Emirates",
  "name": "Dubai",
  "timezone": "Asia/Dubai",
}
`;

exports[`utils should convert values to cities 2`] = `
Array [
  Object {
    "country": "United Arab Emirates",
    "name": "Dubai",
    "timezone": "Asia/Dubai",
  },
  Object {
    "country": "United Arab Emirates",
    "name": "Dubai",
    "timezone": "Asia/Dubai",
  },
]
`;

exports[`utils should generate city options 1`] = `
Object {
  "city": Object {
    "country": "United Arab Emirates",
    "name": "Dubai",
    "timezone": "Asia/Dubai",
  },
  "label": "Dubai, United Arab Emirates",
  "value": "Dubai, United Arab Emirates",
}
`;
